test_train.py

```diff
@@ -0,0 +1,158 @@
+
+    string = f"""
+     {config_name}: &{config_name}
+       loss: 'l2'
+       lr: 5E-4
+       scheduler: 'CosineAnnealingLR'
+       num_data_workers: 4
+       dt: 1 # how many timesteps ahead the model will predict
+       n_history: 0 #how many previous timesteps to consider
+       prediction_type: 'iterative'
+       prediction_length: {prediction_length} #applicable only if prediction_type == 'iterative'
+       n_initial_conditions: 1 #applicable only if prediction_type == 'iterative'
+       ics_type: "default"
+       save_raw_forecasts: !!bool True
+       save_channel: !!bool False
+       masked_acc: !!bool False
+       maskpath: None
+       perturb: !!bool False
+       add_grid: !!bool False
+       N_grid_channels: 0
+       gridtype: 'sinusoidal' #options 'sinusoidal' or 'linear'
+       roll: !!bool False
+       max_epochs: 1
+       batch_size: 2
+
+       #afno hyperparams
+       num_blocks: 8
+       nettype: 'afno'
+       patch_size: 8
+       width: 56
+       modes: 32
+       #options default, residual
+       target: 'default'
+       in_channels: [0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19]
+       out_channels: [0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19] #must be same as in_channels if prediction_type == 'iterative'
+       normalization: 'zscore' #options zscore (minmax not supported)
+       train_data_path: '{train_data_path}'
+       valid_data_path: '{valid_data_path}'
+       inf_data_path: '{inf_data_path}'
+       exp_dir: '{results_dir}'
+       time_means_path:   '{time_means_path}'
+       global_means_path: '{global_means_path}'
+       global_stds_path:  '{global_stds_path}'
+
+       orography: !!bool False
+       orography_path: None
+
+       log_to_screen: !!bool True
+       log_to_wandb: !!bool True
+       save_checkpoint: !!bool True
+
+       enable_nhwc: !!bool False
+       optimizer_type: 'FusedAdam'
+       crop_size_x: None
+       crop_size_y: None
+
+       two_step_training: !!bool False
+       plot_animations: !!bool False
+
+       add_noise: !!bool False
+       noise_std: 0
+    """
+
+    with tempfile.NamedTemporaryFile(mode='w', delete=False, suffix='.yaml') as f:
+        f.write(string)
+        return f.name
+
+def _save_to_tmpfile(data, dir, filetype='h5'):
+    suffices = {'h5': '.h5', 'npy': '.npy'}
+    if filetype not in suffices:
+        raise ValueError(f'Unknown save format {filetype}')
+
+    with tempfile.NamedTemporaryFile(dir=dir, mode='w', delete=False, suffix=suffices[filetype]) as f:
+        if filetype == 'h5':
+            with h5py.File(f.name, 'w') as hf:
+                hf.create_dataset('fields', data=data)
+        elif filetype == 'npy':
+            np.save(f.name, data)
+        else:
+            raise ValueError(f'Unknown save format {filetype}')
+        return f.name
+
+def test_train_runs_era5():
+    """Make sure that training runs without errors."""
+
+    # TODO(gideond) parameterize
```

Comment:
I think this can be left for a later date. Running on multiple random seeds is unlikely to break the runtime and changing data dimensions is likely to require significantly more work -- but let's discuss this if necessary.